pandas/core/base.py

```diff
@@ -292,12 +292,10 @@ def _try_aggregate_string_function(self, arg, *args, **kwargs):
 
         f = getattr(np, arg, None)
         if f is not None:
-            try:
+            if hasattr(self, "__array__"):
```

Comment:
Do we need to account for `__array_ufunc__` or `__array_function__` here as well? @TomAugspurger might want to take a look

Reply:
IIUC, this change is OK since we're checking if `self` has `__array__`, which our Index / Series objects have (in addition to array_ufunc)